packages/headless/src/composables/menuButton/createMenuButton.ts

```diff
@@ -11,139 +11,144 @@ type CreateMenuButtonOptions = {
 /**
  * Based on https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/
  */
-export const createMenuButton = createBuilder((options: CreateMenuButtonOptions) => {
-  const rootId = useId();
-  const menuId = useId();
-  const menuRef = createElRef<HTMLElement>();
-  const buttonId = useId();
+export const createMenuButton = createBuilder(
+  (options: CreateMenuButtonOptions, trigger?: Readonly<Ref<string>>) => {
```

Comment:
Please add "trigger" to the  `CreateMenuButtonOptions`. You can handle it equivalent to the `isExpanded` prop and option.
Also instead of `string` use the specific types of `hover` and `click`.

```suggestion
  (options: CreateMenuButtonOptions) => {
```